packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts

```diff
@@ -0,0 +1,58 @@
+import * as path from 'path';
+import { projectPaths, workspaceUtils } from '../../../src/';
+
+jest.mock('@salesforce/core', () => {
+  return {
+    Global: {
+      SFDX_STATE_FOLDER: '.sfdx'
+    }
+  };
+});
+
+describe('test project paths', () => {
+  const FAKE_WORKSPACE = '/here/is/a/fake/path/to/';
+
+  describe('test stateFolder', () => {
+    const FAKE_STATE_FOLDER = '.sfdx';
+    let getRootWorkspacePathStub: jest.SpyInstance;
+    let hasRootWorkspaceStub: jest.SpyInstance;
+
+    beforeEach(() => {
+      getRootWorkspacePathStub = jest.spyOn(workspaceUtils, 'getRootWorkspacePath').mockReturnValue(FAKE_WORKSPACE);
+    });
+
+    it('is defined', () => {
+      expect(projectPaths.stateFolder).toBeDefined();
+    });
+
+    it('returns path to the state folder if the project has a root workspace', () => {
+      hasRootWorkspaceStub = jest.spyOn(workspaceUtils, 'hasRootWorkspace').mockReturnValue(true);
+      expect(projectPaths.stateFolder()).toEqual(path.join(FAKE_WORKSPACE, FAKE_STATE_FOLDER));
+    });
+
+    it('returns path to the state folder if the project does not have a root workspace', () => {
+      hasRootWorkspaceStub = jest.spyOn(workspaceUtils, 'hasRootWorkspace').mockReturnValue(false);
```

Comment:
I'm move the `hasRootWorkspaceStub = jest.spyOn(workspaceUtils, 'hasRootWorkspace')` to the beforeEach since you use it in multiple tests and just have 
`hasRootWorkspaceStub.mockReturnValue(false)` in each test. Makes for easier reading